lib/puppet/face/config.rb

```diff
@@ -106,6 +126,9 @@
     EOT
 
     when_invoked do |name, value, options|
+
+      options[:section] = options[:section].to_s if options[:section].is_a? Symbol #Value was left as default - set to default string
```

Comment:
It's pretty common to leave off the `is_? Symbol` check in cases like this, as both `String`, and `Symbol` respond to `to_s`, and in the case of `String` it returns itself so it's a no-op.